File: tests/unit/X2Mcp.Language.Go.Tests/Fixtures/methods.go
package fixtures

type Calculator struct{}

type CalculatorGeneric[T any] struct{}

func (c Calculator) Add(a, b int) int {
	return a + b
}

func (c *Calculator) Multiply(a, b int) int {
	return a * b
}

func (c CalculatorGeneric[T]) Generic(a int) int {
	return a
}

func (c Calculator) hidden() int {
	return 0
}

type internalCalculator struct{}

func (c internalCalculator) PublicButInternalType() int {
	return 1
}

File: tests/unit/X2Mcp.Language.Rust.Tests/Fixtures/methods.rs
pub struct Calculator;

impl Calculator {
    pub fn add(&self, a: i32, b: i32) -> i32 {
        a + b
    }

    pub fn multiply(&self, a: i32, b: i32) -> i32 {
        a * b
    }

    fn hidden(&self) -> i32 {
        0
    }
}

struct InternalCalculator;

impl InternalCalculator {
    pub fn public_but_internal_type(&self) -> i32 {
        1
    }
}
